docs/DeveloperGuide.md

```diff
@@ -4,10 +4,150 @@
+- Returns the`Command`object that can then be executed.
+
+We pass in the`itemList`to the`Command`object instead of using a global variable to ease testing. This way, we can 
+add, delete and update`Expenditure`entries in a new`itemList`during testing without affecting the actual`itemList`.
+
+### Command component 
+The source can be found in [command](https://github.com/AY2122S2-CS2113T-T10-4/tp/blob/master/src/main/java/seedu/mindmymoney/command)
+
+![command_diagram](images/command_diagram.png)
+<br/> Fig 5 - Command Diagram
+
+The Command component consists of `Command` abstract class, `AddCommand`, `ByeCommand`, `DeleteCommand`, `HelpCommand`
+, `ListCommand`and `UpdateCommand` that extends the `Command` class.
+
+The Command component:
+- Provides all the Command classes which can be instantiated by `Parser.parseCommand()`. The Command objects can then be 
+executed. Only 1 Command object can be created.
+- Includes a `.executeCommand()` method in each Command classes which performs the relevant command and throws 
+exceptions if an error is encountered. The error is then handled.
+
+### Storage component 
+The source can be found in [Storage.java](https://github.com/AY2122S2-CS2113T-T10-4/tp/blob/master/src/main/java/seedu/mindmymoney/Storage.java)
+
+![storage_diagram](images/storage_diagram.png)
+<br/> Fig 6 - Storage Diagram
+
+The Storage component consists of `Storage` class.
+
+The Storage component:
+- `MMM` class initialises a `Storage` object upon start up. The `Storage` class consists of 
+`load()` and `save()` methods. 
+- Concurrently, `MMM` will call the `load()` method and load any data that is stored on the hard disk.
+- `MMM` calls the `save()` method and stores remaining data onto the hard disk when the program exits.
+
+
+## Implementation
+This section describes some noteworthy details on how certain features are implemented.
+
+### Add Command Feature
+#### Current Implementation
+The sequence diagram below shows the interactions of different subcomponents of the system when adding an expenditure
+to the list.  
+![add_command_sequence_diagram](images/Add_Command_Sequence_Diagram.png)  
+A key functionality of MindMyMoney is the ability to add and track user expenditure. A user can add in a new expenditure
+by specifying the payment type, the category, the description of the item, the cost of the item and the date and the
+month purchased.
+
+1. The `Parser` component parses user input and returns the new `AddCommand` object to the
+   `MindMyMoney`.
+2. `AddCommand` instantiates `addInput`, `expenditureList` and `creditCardList`.
+3. The application invokes `Addcommand.execute` to execute user instruction.
+4. During the execution, `Addcommand.execute` will parse through user input to obtain the `EXPENDITURE`, `CATEGORY`,
+   `DESCRIPTION`, `AMOUNT` and `TIME` fields.
+5. The `Addcommand` object instantiates a new `Expenditure` object with the aforementioned 5 fields and adds them
```

Comment:
Perhaps there should be another line to describe what the AddCommandInputTest#test(...) method does?